### PR TITLE
Upgrade JDK to 8 refer #1863

### PR DESCRIPTION
**Pull request by smadasu**

### CODE REVIEW

**Comment by smadasu** [comment]
This build fails and succeeds intermittently on my local sand box. I don't know why

File: src/main/java/redis/clients/jedis/params/SetParams.java [hunk 1]
```diff
@@ -71,11 +71,13 @@ public SetParams xx() {
 
     if (contains(EX)) {
       byteParams.add(SafeEncoder.encode(EX));
-      byteParams.add(SafeEncoder.encode(String.valueOf(getParam(EX))));
+      final Object param = getParam(EX);
+      byteParams.add(SafeEncoder.encode(String.valueOf(param)));
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/params/SetParams.java]: How are these changes related to JDK 8?

**Comment by sazzad16** [comment]
@gkorland About https://github.com/xetorthio/jedis/pull/1864#pullrequestreview-173921091

Those changes are related to JDK 8. The [job log](https://travis-ci.org/xetorthio/jedis/jobs/426423621) contains related exceptions. Here https://stackoverflow.com/questions/23155907/classcast-error-java-7-vs-java-8 are some details about the case.

However, IMHO, we should come up with a generic solution for these situations. Otherwise, there would be redundant codes over and over again.

@marcosnils ping!

File: src/main/java/redis/clients/jedis/params/SetParams.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/params/SetParams.java]: Resolved in #2019

**Comment by sazzad16** [comment]
Added same milestone as #2040

**Comment by sazzad16** [comment]
Resolved by #2396
